pygbe/main.py

```diff
@@ -355,14 +367,27 @@ def main(argv=sys.argv, log_output=True, return_output_fname=False,
     print('-'*30)
     print('Total setup time   : {}s'.format(setup_time))
 
-    tic = time.time()
+
+    #   Check if there is a complex dielectric
+    complexDiel = 0
+    for f in field_array:
+        if numpy.iscomplexobj(f.E):
+            complexDiel = 1
+
 
     ### Solve
+    tic = time.time()
+
     print('Solve')
-    phi = numpy.zeros(param.Neq)
+    # Initializing phi dtype according to the problem we are solving.
+    if complexDiel == 1:
+        phi = numpy.zeros(param.Neq, type(f.E))
```

Comment:
I don't care so much about number of lines so much as readability. 

But I _really_ don't like that `f.E` is going to depend on the value of the last array in `field_array`.  That's a little too random. It might be best to also use an `any` when we define the datatype below.